openpype/hosts/fusion/plugins/create/create_workfile.py

```diff
@@ -0,0 +1,129 @@
+import collections
+
+import qtawesome
+
+import openpype.hosts.fusion.api as api
+from openpype.client import get_asset_by_name
+from openpype.pipeline import (
+    AutoCreator,
+    CreatedInstance,
+    legacy_io,
+)
+
+
+def flatten_dict(d, parent_key=None, separator="."):
+    items = []
+    for key, v in d.items():
+        new_key = parent_key + separator + key if parent_key else key
+        if isinstance(v, collections.MutableMapping):
+            items.extend(flatten_dict(v, new_key, separator=separator).items())
+        else:
+            items.append((new_key, v))
+    return dict(items)
+
+
+class FusionWorkfileCreator(AutoCreator):
+    identifier = "workfile"
+    family = "workfile"
+    label = "Workfile"
+
+    default_variant = "Main"
+
+    create_allow_context_change = False
+
+    data_key = "openpype.workfile"
+
+    def collect_instances(self):
+
+        comp = api.get_current_comp()
+        data = comp.GetData(self.data_key)
+        if not data:
+            return
+
+        instance = CreatedInstance(
+            family=self.family,
+            subset_name=data["subset"],
+            data=data,
+            creator=self
+        )
+        instance.transient_data["comp"] = comp
+        instance.transient_data["tool"] = None
+
+        self._add_instance_to_context(instance)
+
+    def update_instances(self, update_list):
+        for update in update_list:
+            instance = update.instance
+            comp = instance.transient_data["comp"]
+            if not hasattr(comp, "SetData"):
+                # Comp is not alive anymore, likely closed by the user
+                self.log.error("Workfile comp not found for existing instance."
+                               " Comp might have been closed in the meantime.")
+                continue
+
+            # TODO: It appears sometimes this could be 'nested'
+            # Get the new values after the changes by key, ignore old value
+            new_data = {
+                key: new for key, (_old, new) in update.changes.items()
+            }
+            self._imprint(comp, new_data)
+
+    def create(self, options=None):
+
+        comp = api.get_current_comp()
+        if not comp:
+            self.log.error("Unable to find current comp")
+            return
+
+        # TODO: Is this really necessary?
+        # Force kill any existing "workfile" instances
+        for instance in self.create_context.instances:
+            if instance.family == self.family:
+                self.log.debug(f"Removing instance: {instance}")
+                self._remove_instance_from_context(instance)
```

Comment:
I'm still not entirely sure if the AutoCreator has to implement both a `create()` and a `collect_instances()`. It feels like both are trying to do the same thing - which is automatically give me the instance from the scene.

This confusing mixture in responsibility makes it so that I still don't understand what should be in `collect_instances` and what should be in `create` and why?

In my logic here I'm force killing it - in other Workfile creators or the documentation example it tries to update it in-place. Why would we care to do one over the other?